tests/sklearn/test_sklearn_models.py

```diff
@@ -1871,3 +1921,87 @@ def test_rounding_consistency_for_regular_models(
         metric,
         is_weekly_option,
     )
+
+
+@pytest.mark.parametrize("model_class, parameters", get_sklearn_tree_models_and_datasets())
+@pytest.mark.parametrize("n_bits", [2, 5, 10])
+def test_fhe_sum_for_tree_based_models(
+    model_class,
+    parameters,
+    n_bits,
+    load_data,
+    is_weekly_option,
+    verbose=True,
+):
+    """Test that the tree ensembles' output are the same with and without the sum in FHE."""
+
+    if verbose:
+        print("Run check_fhe_sum_for_tree_based_models")
+
+    model = instantiate_model_generic(model_class, n_bits=n_bits)
+
+    x, y = get_dataset(model_class, parameters, n_bits, load_data, is_weekly_option)
+
+    predict_method = (
+        model.predict_proba if is_classifier_or_partial_classifier(model) else model.predict
+    )
+
+    check_fhe_sum_for_tree_based_models(
+        model,
+        x,
+        y,
+        predict_method,
+        is_weekly_option,
+    )
+
+
+@pytest.mark.parametrize(
+    "n_bits, error_message",
+    [
+        (0, "n_bits must be a non-null, positive integer"),
+        (-1, "n_bits must be a non-null, positive integer"),
+        ({"op_leaves": 2}, "The key 'op_inputs' is mandatory"),
+        (
+            {"op_inputs": 4, "op_leaves": 2, "op_weights": 2},
+            "Invalid keys in 'n_bits' dictionary. Only 'op_inputs' \\(mandatory\\) and 'op_leaves' "
+            "\\(optional\\) are allowed",
+        ),
+        (
+            {"op_inputs": -2, "op_leaves": -5},
+            "All values in 'n_bits' dictionary must be non-null, positive integers",
+        ),
+        ({"op_inputs": 2, "op_leaves": 5}, "'op_leaves' must be less than or equal to 'op_inputs'"),
+        (0.5, "n_bits must be either an integer or a dictionary"),
+    ],
+)
+@pytest.mark.parametrize("model_class", _get_sklearn_tree_models())
+def test_invalid_n_bits_setting(model_class, n_bits, error_message):
+    """Check if the model instantiation raises an exception with invalid 'n_bits' settings."""
+
+    with pytest.raises(ValueError, match=f"{error_message}. Got '{type(n_bits)}' and '{n_bits}'.*"):
+        instantiate_model_generic(model_class, n_bits=n_bits)
+
+
+@pytest.mark.parametrize("n_bits", [5, {"op_inputs": 5}, {"op_inputs": 2, "op_leaves": 1}])
```

Comment:
could you add a fixme indicating that this test should be extended to all built-ins please ? using this issue :  https://github.com/zama-ai/concrete-ml-internal/issues/4234